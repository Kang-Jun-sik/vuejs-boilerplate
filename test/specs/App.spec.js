import {
  shallow,
  mount,
} from 'vue-test-utils';
import List from '@/App';
import {
  createRenderer,
} from 'vue-server-renderer';

describe('App.vue', () => {
  it('renders li for each item in props.items', () => {
    const items = ['', ''];
    const wrapper = shallow(List, {
      propsData: { items },
    });
    expect(wrapper.findAll('li')).toHaveLength(items.length);
  });

  it('matches snapshot', () => {
    const items = ['item 1', 'item 2'];
    const renderer = createRenderer();
    const wrapper = mount(List, {
      propsData: { items },
    });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
