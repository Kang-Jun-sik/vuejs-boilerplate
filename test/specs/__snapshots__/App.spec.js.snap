// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
<div id="app">
    <h1>Hello Vuejs</h1>
    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis repellat quaerat, ut, incidunt culpa repudiandae accusamus dolorem obcaecati error corrupti vero, eveniet qui explicabo laborum possimus. Maiores minus molestias provident!</p>
    <ul>
        <li>
            item 1
        </li>
        <li>
            item 2
        </li>
    </ul>
    <!---->
</div>
`;
